src/command/restore/restore.c

```diff
@@ -1232,11 +1279,14 @@ restoreSelectiveExpression(Manifest *manifest)
                     includeDb = varStrForce(VARUINT(db->id));
                 }
 
+                // Error if the db id is in the exclude list
+                if(strLstExists(excludeDbIdList, includeDb))
+                    THROW_FMT(DbInvalidError, "database to include '%s' is in the exclude list", strZ(includeDb));
+
                 // Error if the db is a system db
-                if (cvtZToUInt64(strZ(includeDb)) < PG_USER_OBJECT_MIN_ID)
+                if (cvtZToUInt64(strZ(includeDb)) < PG_USER_OBJECT_MIN_ID || strLstExists(systemDbIdList, includeDb))
```

Comment:
yaeh, adding all the system dbs to the list simplifies a lot of conditions :-)

Reply:
Actually, this one is pretty tricky. 
It breaks the MockAllTest because we don't generate the `[db]` section in the manifest.
I tried to add (line 1077 - "error on system id"):
```
$oManifest{&MANIFEST_SECTION_BACKUP_DB}{"template1"}{&MANIFEST_KEY_DB_ID} = 1;
$oManifest{&MANIFEST_SECTION_BACKUP_DB}{"template1"}{&MANIFEST_KEY_DB_LAST_SYSTEM_ID} = 12168;
```
But no luck. I'm not comfortable enough with the perl tests at this point. Would you have any idea to fix this test?

Many thanks in advance :-)

Reply:
No worries. I should have some time tomorrow to take a look.

Reply:
I pulled and began working on the branch in my own repository (https://github.com/cmwshang/pgbackrest/tree/pgstef-issues-986) so you can compare. I haven't yet completed the tests but will let you know when I do. Thanks again.

Reply:
Seems the code cover this exception now:

```
if (cvtZToUInt64(strZ(dbId)) < PG_USER_OBJECT_MIN_ID)
                   strLstAddIfMissing(systemDbIdList, dbId);
```